product_listings_manager/__init__.py

```diff
@@ -18,7 +18,7 @@ def getProductInfo(*a, **kw):
     except Exception as e:
         err = traceback.print_exc()
         app.logger.error(err)
-        raise Fault(e)
+        raise Fault(1, repr(e))
```

Comment:
It's better to just log unexpected exceptions and keep HTTP 500. Otherwise, I think it could leak some some unexpected information to user.

Reply:
That's a good point. I've amended this to just give the `type()` instead of the full `repr()`. Does that sound ok?

Reply:
It would be nice to always pass some user-friendly error message (wrong arguments, authentication issue etc). For the really unexpected exceptions, it would be something like "An unexpected error has occurred, please file an issue".

Reply:
Oh, got it. I've amended this again to return a basic string in the truly unexpected case.